eZ/Publish/Core/MVC/Symfony/FieldType/ImageAsset/ParameterProvider.php

```diff
@@ -33,17 +41,52 @@ public function __construct(ContentService $contentService)
     public function getViewParameters(Field $field): array
     {
         try {
-            $contentInfo = $this->contentService->loadContentInfo(
+            $contentInfo = $this->loadContentInfo(
                 $field->value->destinationContentId
             );
 
-            return [
-                'available' => !$contentInfo->isTrashed(),
-            ];
-        } catch (NotFoundException | UnauthorizedException $exception) {
+            if (!$this->userHasPermissions($contentInfo)) {
+                return [
+                    'available' => false,
+                ];
+            }
+
+            return ['available' => !$contentInfo->isTrashed()];
+        } catch (NotFoundException $exception) {
             return [
                 'available' => false,
             ];
         }
     }
+
+    /**
+     * @param int $id
+     * @return \eZ\Publish\API\Repository\Values\Content\ContentInfo
+     * @throws \Exception
+     */
+    private function loadContentInfo(int $id): ContentInfo
+    {
+        return  $this->repository->sudo(
+            function (Repository $repository) use ($id) {
+                return $repository->getContentService()->loadContentInfo($id);
```

Comment:
Not a change request but rather a concern:

This is not a really good idea. Over time it might become forgotten security hole. 
I get that you're repeating that permission check in `userHasPermissions`, but who will guarantee that this set of permissions will be always used for `loadContentInfo`? What if `loadContentInfo` internal logic changes so it's not readable due to another reason? While it might be correct now, there is no way to track any changes in the future.

I'd say that maybe we need to make Repository aware that we're loading for embedding. 
```php
interface ContentService {
    // ...
    public function loadContentInfo(int $contentId, bool $loadForEmbedding = false);
```
But maybe I'm overthinking this :P

WDYT @adamwojs @andrerom @Nattfarinn @mikadamczyk?

Reply:
the solution was proposed by @adamwojs https://github.com/ezsystems/ezpublish-kernel/pull/2513#issuecomment-452115730

Reply:
@alongosz I like your idea but as follow up PR as this need be fixed in several other places as well (e.g. embedding images and content in richtext) 😉

Reply:
Related discussion on `loadContentInfoList(array $contentIds, bool $filterOnUserPermissions = true)`:
https://github.com/ezsystems/ezpublish-kernel/pull/2529